app/controllers/ncr/proposals_controller.rb

```diff
@@ -29,6 +32,46 @@ def create
       end
     end
 
+    def not_approved
+      cart = Cart.find(params[:id])
```

Comment:
Mind making a shared method so there's only a single call to the database between all of the methods?

```ruby
def cart
  @cart ||= Cart.find(params[:id])
end
```